sql/catalyst/src/main/java/org/apache/spark/sql/connector/catalog/TableInfo.java

```diff
@@ -29,6 +30,7 @@ public class TableInfo {
   private final Column[] columns;
   private final Map<String, String> properties;
   private final Transform[] partitions;
+  private final Constraint[] constraints;
```

Comment:
We need to add a new capability to `TableCatalogCapability`. Right now, if Spark starts to pass constraints, it may be ignored by the connector. Therefore, we need a new capability so that connectors can tell Spark that they support constraints. Otherwise, they may simply swallow and ignore them.

Reply:
+1

Reply:
One question, as there are different kind of constraints, if a catalog claims it support the new constraints capability, does it must support all kinds of constraints? Do we need to have fine-grained capability of constraints?

Reply:
@viirya good suggestion. It makes more sense to have a capability for each constraint.
@aokolnychyi WDYT?

Reply:
We just need a capability that confirms the connector will look into passed constraints. It may still reject some of them if it doesn't support a particular constraint. The same applies today to default values. We have a single flag but connectors can still discard some expressions.